### ISSUE TITLE
[regression?] Not able to create a vk::UniqueSurfaceKHR from VkSurfaceKHR in new SDK

### ISSUE THREAD

**Issue by Ybalrid** [KhronosGroup/Vulkan-Hpp#467]
Hi,

I am not sure that this is the right place to submit that issue, but, using `vulkan.hpp`, when upgrading from LunarG SDK **1.1.126.0** to **1.1.130.0**, I can no longer construct  a `vk::UniqueHandleKHR` from a `VkSurfaceKHR`, by passing it the C object and the vk::Instance in use.

Doing the following is a stardard way of delegating the code that create the SurfaceKHR to a library like SDL2 or GLFW that will do *the right thing* for the current platform/windowing system (as they also created the window, and provided you the list of required instance extensions).

Basically, the following does work with 1.1.126 (and earlier versions) but not 1.1.130:

```cpp
vk::UniqueSurfaceKHR createSurfaceUnique(vk::Instance instance)
{
  VkSurfaceKHR surface;
  CallToSDL2orGLFWthatCreateASurfaceKHR(m_window, instance, &surface); 
  retrun vk::UniqueSurfaceKHR(surface, instance);
}
```

**Comment by zisner** [comment]
Hello,

I've run into the same issue and have been testing the lastest release across a few platforms.

Here's the code that worked in older releases, but not in 1.1.130.0:

```
void VulkanBase::CreatePresentationSurface() {

    VkSurfaceKHR tmpSurface;
    VkResult err = glfwCreateWindowSurface(Vulkan.Instance, Window.WindowPtr, nullptr, &tmpSurface);

    if (err) {
        std::runtime_error( "Window surface creation failed" );
    }

    
    Vulkan.PresentationSurface = vk::UniqueSurfaceKHR(tmpSurface, Vulkan.Instance);

}
```

Here are the errors I've been getting:

Windows 10:
```
[build] ..\Include\Core\c_VulkanBase.cpp(202): error C2440: '<function-style-cast>': cannot convert from 'initializer list' to 'vk::UniqueSurfaceKHR'
[build] ..\Include\Core\c_VulkanBase.cpp(202): note: No constructor could take the source type, or constructor overload resolution was ambiguous
```

macOS Mojave:

```
[build] /home/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-linux-x86_64-1.1.130.0/1.1.130.0/x86_64/include/vulkan/vulkan.hpp:609:5: note: candidate: vk::UniqueHandle<Type, Dispatch>::UniqueHandle(vk::UniqueHandle<Type, Dispatch>&&) [with Type = vk::SurfaceKHR; Dispatch = vk::DispatchLoaderStatic]
[build]      UniqueHandle( UniqueHandle && other ) VULKAN_HPP_NOEXCEPT
[build]      ^~~~~~~~~~~~
[build] /home/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-linux-x86_64-1.1.130.0/1.1.130.0/x86_64/include/vulkan/vulkan.hpp:609:5: note:   candidate expects 1 argument, 2 provided
[build] /home/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-linux-x86_64-1.1.130.0/1.1.130.0/x86_64/include/vulkan/vulkan.hpp:602:14: note: candidate: vk::UniqueHandle<Type, Dispatch>::UniqueHandle(const Type&, const Deleter&) [with Type = vk::SurfaceKHR; Dispatch = vk::DispatchLoaderStatic; vk::UniqueHandle<Type, Dispatch>::Deleter = vk::ObjectDestroy<vk::Instance, vk::DispatchLoaderStatic>]
[build]      explicit UniqueHandle( Type const& value, Deleter const& deleter = Deleter() ) VULKAN_HPP_NOEXCEPT
[build]               ^~~~~~~~~~~~
[build] /home/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-linux-x86_64-1.1.130.0/1.1.130.0/x86_64/include/vulkan/vulkan.hpp:602:14: note:   no known conversion for argument 2 from ‘vk::Instance’ to ‘const Deleter& {aka const vk::ObjectDestroy<vk::Instance, vk::DispatchLoaderStatic>&}’
[build] /home/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-linux-x86_64-1.1.130.0/1.1.130.0/x86_64/include/vulkan/vulkan.hpp:597:5: note: candidate: vk::UniqueHandle<Type, Dispatch>::UniqueHandle() [with Type = vk::SurfaceKHR; Dispatch = vk::DispatchLoaderStatic]
[build]      UniqueHandle()
[build]      ^~~~~~~~~~~~
[build] /home/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-linux-x86_64-1.1.130.0/1.1.130.0/x86_64/include/vulkan/vulkan.hpp:597:5: note:   candidate expects 0 arguments, 2 provided
```

Ubuntu 18.04:

```
[build] /Users/DillonRidgeComputing/GPU-Script/QA/Tests_2019_12_31/TestPublicRelease_2019_12_31/Include/Core/c_VulkanBase.cpp:202:38: error: no matching constructor for initialization of 'vk::UniqueSurfaceKHR' (aka 'UniqueHandle<vk::SurfaceKHR, ::vk::DispatchLoaderStatic>')
[build]         Vulkan.PresentationSurface = vk::UniqueSurfaceKHR(tmpSurface, Vulkan.Instance);
[build]                                      ^                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build] /Users/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-macos-1.1.130.0/macOS/include/vulkan/vulkan.hpp:602:14: note: candidate constructor not viable: no known conversion from 'vk::Instance' to 'const vk::UniqueHandle<vk::SurfaceKHR, vk::DispatchLoaderStatic>::Deleter' (aka 'const ObjectDestroy<vk::Instance, vk::DispatchLoaderStatic>') for 2nd argument
[build]     explicit UniqueHandle( Type const& value, Deleter const& deleter = Deleter() ) VULKAN_HPP_NOEXCEPT
[build]              ^
[build] /Users/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-macos-1.1.130.0/macOS/include/vulkan/vulkan.hpp:607:5: note: candidate constructor not viable: requires 1 argument, but 2 were provided
[build]     UniqueHandle( UniqueHandle const& ) = delete;
[build]     ^
[build] /Users/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-macos-1.1.130.0/macOS/include/vulkan/vulkan.hpp:609:5: note: candidate constructor not viable: requires single argument 'other', but 2 arguments were provided
[build]     UniqueHandle( UniqueHandle && other ) VULKAN_HPP_NOEXCEPT
[build]     ^
[build] /Users/DillonRidgeComputing/GPU-Script/Builds/Vulkan_Builds/vulkansdk-macos-1.1.130.0/macOS/include/vulkan/vulkan.hpp:597:5: note: candidate constructor not viable: requires 0 arguments, but 2 were provided
[build]     UniqueHandle()
[build]     ^
[build] 1 error generated.
```


I've also found that if you don't pass in the vulkan instance to `vk::UniqueSurfaceKHR`, then it compiles and runs. However, upon exiting the application, it fails an assertion in the ObjectDestroy class in vulkan.hpp. This assert appears to have been added in the 1.1.130.0 release...

```
Assertion failed: m_owner && m_dispatch, file C:\VulkanSDK\1.1.130.0\Include\vulkan/vulkan.hpp, line 2713
```

Any ideas would be greatly appreciated. This is preventing our application from working with the latest Vulkan release. Thanks a bunch!

**Comment by mtavenrath** [comment]
@Ybalrid @zisner The regression got introduced by the fix for issue #443 where we lost two default arguments for the constructor.  For the example I verified that it does work now and I added a test which will prevent a that this feature breaks again.

Feel free to copy https://raw.githubusercontent.com/KhronosGroup/Vulkan-Hpp/master/vulkan/vulkan.hpp over your Vulkan SDK 1.1.130 header until the fix has been picked up by the next SDK release.

**Comment by Ybalrid** [comment]
@zisner The assert failing (or segfault when not in debug) is expected if a UniqueHandle doesn't have a "deleter". It will then de-reference `nullptr` when getting out of scope, exploding in your face (instead of liberating the resource RAII style).

@mtavenrath Thanks for pushing the fix! Will update the headers locally and test it

**Comment by Ybalrid** [comment]
Updated header does indeed work!

**Comment by mtavenrath** [comment]
Thanks for the feedback. Closing.

**Comment by 9291Sam** [comment]
For anyone who is stumbling upon this now, you can just pass the instance as a deleter and it'll work just fine. 
```cpp
vk::UniqueSurfaceKHR uniqueSurface {vk::Surface {surface}, instance};
```
